exercise-3.sql

```diff
@@ -0,0 +1,107 @@
+--    Info:  grouped by department id and job id ordered by department id and max salary
+select d.department_name,e.job_id,max(e.salary) as max_salary,min(e.salary) as min_salary,avg(e.salary) as avg_salary from employees e inner join departments d on e.department_id = e.department_id group by e.job_id,d.department_name order by d.department_name,max_salary; 
+
+-- 5) Write a SQL query to find the total salary of employees whose country_id is ‘US’ excluding whose first name is Nancy  
+select l.country_id,sum(e.salary) as total_salary from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id where l.country_id = 'US' and e.first_name != 'Nancy' group by l.country_id;
+
+
+-- 6) Fetch max salary, min salary and avg salary by job id and department id but only for folks who worked in more than one role(job) in a department.
+select d.department_name,e.job_id,max(e.salary) as max_salary,min(e.salary) as min_salary,avg(e.salary) as avg_salary from employees e inner join departments d on e.department_id = e.department_id inner join job_history j on e.employee_id=j.employee_id group by e.job_id,d.department_name order by d.department_name,max_salary; 
+
+select max(salary) , min(salary) , avg(salary), e.department_id from employees e inner join  job_history j on e.employee_id = j.employee_id group by e.department_id, e.job_id;
+
+
+-- 7) Display the employee count in each department and also in the same result.  
+select ifnull(d.department_name,'-')as Department_Name,count(*) as Total from employees e inner join departments d on e.department_id = d.department_id group by d.department_name order by Total;
+
+
+-- 8)  Display the jobs held and the employee count. 
+--     Hint: every employee is part of at least 1 job 
+--     Hint: use the previous questions answer
+--     Sample
+--     JobsHeld EmpCount
+--     1	    100
+--     2	    4
+
+select m.jobs_held , count(emp_id) as emp_count from(select e.employee_id as emp_id , count(*) as jobs_held from(select employee_id,job_id from employee union all select employee_id ,job_id from job_history order by employee_id) as e group by e.employee_id) as m group by jobs_held order by jobs_held;
+
+-- 9) Display average salary by department and country.
+select d.department_id,l.country_id,avg(e.salary) from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id group by d.department_id,l.country_id order by d.department_id,l.country_id;
+
+-- 10)	Display manager names and the number of employees reporting to them by countries (each employee works for only one department, and each department belongs to a country)
+select d.manager_id,l.country_id,count(*) from employees e inner join departments d on e.department_id=d.department_id inner join 
+locations l on l.location_id=d.location_id group by d.manager_id,l.country_id;
+
```

Comment:
Does this fetch the expected result ?
Some joins and logic are missing . Recheck your results